ha/core/event_manager/event_manager.py

```diff
@@ -0,0 +1,159 @@
+                        # Delete the component if it exists in the list.
+                        if component in component_list:
+                            component_list.remove(component)
+                        if len(component_list) == 0:
+                            self._confstore.delete(component_list_key)
+                        else:
+                            self._confstore.update(component_list_key, json.dumps(component_list))
+                        Log.debug(f"Updated components {component_list} subscribed to event {event}")
+
+                # TODO: Before below, check if any subscriptions present for this component
+                # Check and remove message type (Queue) if it exists for this component
+                message_bus_admin = MessageBusAdmin(const.HA_MESSAGE_BUS_ADMIN)
+                message_type = const.EVENT_MGR_MESSAGE_TYPE.replace("<component_id>", component)
+                message_type_list = message_bus_admin.list_message_types()
+                if message_type in message_type_list:
+                    message_bus_admin.deregister_message_type(message_types = [message_type])
+                # Remove message type key from confstore
+                message_type_key = const.EVENT_MGR_MESSAGE_TYPE_KEY.replace("<component_id>", component)
+                if self._confstore.key_exists(message_type_key):
+                    self._confstore.delete(message_type_key)
+                Log.debug(f"Removed message type {message_type} for component {component}")
+                Log.info(f"Unsubscribed component {component}")
+        except Exception as e:
+            Log.error(f"{component} unsubscription failed, Error: {e}")
+            raise UnsubscriptionException(f"{component} unsubscription failed, Error: {e}")
+
+    def publish(self, event: ActionEvent) -> None:
+        """
+        Publish event.
+        Args:
+            event (ActionEvent): Action event.
+        """
+        try:
+            component_list = []
+            # Run through list of components subscribed for this event and send event to each of them
+            component_list_key = const.EVENT_COMPONENT_LIST.replace("<event_name>", event.resource_type)
+            if self._confstore.key_exists(component_list_key):
+                component_list_key_val = self._confstore.get(component_list_key)
+                _, value = component_list_key_val.popitem()
+                component_list = json.loads(value)
+            for component in component_list:
+                producer_id = const.EVENT_MGR_PRODUCER_ID.replace("<component_id>", component)
+                message_type_key = const.EVENT_MGR_MESSAGE_TYPE_KEY.replace("<component_id>", component)
+                message_type_key_val = self._confstore.get(message_type_key)
+                _, message_type = message_type_key_val.popitem()
+                message_producer = MessageProducer(producer_id = producer_id,
```

Comment:
Can we saperate message bus and this code saperate
message bus needed many place so want to centralize it

https://github.com/Seagate/cortx-ha/pull/514/files
created class under util for same reason

Reply:
done, using the class available in PR #514